src/components/Toggle/Toggle.props.ts

```diff
@@ -0,0 +1,15 @@
+import { MouseEvent } from 'react';
+
+import type { ToggleConfig } from './Toggle.styles';
+import { HelperTextProps } from '../HelperText';
+
+export type ToggleProps = {
+  isIndeterminate?: boolean;
+  isChecked?: boolean;
+  size?: 'small' | 'large';
+  state?: 'disabled' | undefined;
```

Comment:
I think we can remove `undefined` because it is already covered by optional property expression.
```suggestion
  state?: 'disabled';
```